File: codingTest/programmers/javascript/프린터.js
var solution = (priorities, location) => {
  priorities = priorities.map((e, i) => [i, e]);
  let max = Math.max(...priorities.map(e => e[1]));
  let count = 0;
  while (priorities.length) {
    if (priorities[0][1] != max) priorities.push(priorities.shift());
    else {
      if ((++count, priorities.shift()[0] == location)) return count;
      max = Math.max(...priorities.map(e => e[1]));
    }
  }
};

File: codingTest/programmers/python/프린터.py
from collections import deque
def solution(priorities, location):
    printer = deque([])
    for i, v in enumerate(priorities): printer.append([i, v])
    
    cnt = 0
    popPrint = -1
    while location != popPrint:
        if printer[0][1] == max(printer, key=lambda x: x[1])[1]:
            cnt += 1
            popPrint = printer.popleft()[0]
        else: printer.append(printer.popleft())
    return cnt